testsuite/cluster-test/src/experiments/load_test.rs

```diff
@@ -389,26 +418,35 @@ async fn state_sync_load_test(
     let mut served_txns = 0_u64;
     let mut bytes = 0_u64;
     let mut msg_num = 0_u64;
+    let mut pending = 0_usize;
     while Instant::now().duration_since(task_start) < duration {
-        let msg = state_synchronizer::network::StateSynchronizerMsg::GetChunkRequest(Box::new(
-            chunk_request.clone(),
-        ));
-        bytes += lcs::to_bytes(&msg)?.len() as u64;
-        msg_num += 1;
-        sender.send_to(vfn, msg)?;
-
+        if pending < NETWORK_CHANNEL_SIZE {
+            let msg = state_synchronizer::network::StateSynchronizerMsg::GetChunkRequest(Box::new(
+                chunk_request.clone(),
+            ));
+            bytes += lcs::to_bytes(&msg)?.len() as u64;
+            sender.send_to(vfn, msg)?;
+            pending += 1;
+        }
         // await response from remote peer
-        let response = events.select_next_some().await;
-        if let Event::Message(_remote_peer, payload) = response {
-            if let state_synchronizer::network::StateSynchronizerMsg::GetChunkResponse(
-                chunk_response,
-            ) = payload
-            {
-                // TODO analyze response and update StateSyncResult with stats accordingly
-                served_txns += chunk_response.txn_list_with_proof.transactions.len() as u64;
+        while let Some(response) = events.select_next_some().now_or_never() {
+            if let Some(txs_num) = process_poll_msg(response) {
+                served_txns += txs_num as u64;
+                pending -= 1;
+                msg_num += 1;
             }
         }
     }
+    // wait for all msgs are processed and acks back
+    loop {
+        let response = events.select_next_some().await;
+        if let Some(txs_num) = process_poll_msg(response) {
```

Comment:
no no no, this doesn't work, if it's some different messages coming back, you'll just break the loop. you need to use the pending counter